packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -33,11 +31,49 @@ const { prefix } = settings;
  */
 const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
   const [selectedId, setSelectedId] = useState(menuItems[0].id);
-  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
-
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].name);
+  const [autoMenuItems, setmenuItems] = useState([]);
   useEffect(() => {
+    /**
+     * Set selected id & title
+     *
+     */
+    if (menuItems) {
+      var setSelectedItem = () => {
+        const elems = getElemsInView();
+        const id = elems[0] || menuItems[0].id;
+        const filteredItems = menuItems.filter(menu => {
+          if (id !== 'undefined') {
+            return menu.id == id;
+          }
+        });
+        const title = filteredItems[0].title;
+        setSelectedId(id);
+        setSelectedTitle(title);
+      };
+    }
+    /**
+     * loops into the array of elements and returns the values
+     *
+     * @private
+     * @returns {Array} returns elemenrt name and data title
+     */
+    const _find_menu_items = () => {
```

Comment:
The name of this doesn't follow convention. Should be something like `findMenuItems` instead.

Reply:
Also, this can remain as a private method, but can be outside of the Class (doesn't need to be inside of it).